test/example-types/metricGroups/pivot.js

```diff
@@ -158,6 +158,269 @@ describe('pivot', () => {
+              aggs: {
+                groups: {
+                  range: {
+                    field: 'LineItem.TotalPrice',
+                    ranges: [
+                      { from: '0', to: '500' },
+                      { from: '500', to: '10000' },
+                    ],
+                  },
+                  aggs: {
+                    'pivotMetric-sum-LineItem.TotalPrice': {
+                      sum: { field: 'LineItem.TotalPrice' },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    }
+    let result = await buildQuery(
+      input,
+      testSchemas(['Vendor.City']),
+      () => {} // getStats(search) -> stats(field, statsArray)
+    )
+    expect(result).to.eql(expected)
+    let resultTopLevel = await buildQuery(
+      inputTopLevel,
+      testSchemas(['Vendor.City']),
+      () => {} // getStats(search) -> stats(field, statsArray)
+    )
+    expect(resultTopLevel).to.eql(expected)
+  })
+  it('should buildQuery for fieldValues with drilldown and limited depth', async () => {
+    let input = {
+      key: 'test',
+      type: 'pivot',
+      values: [{ type: 'sum', field: 'LineItem.TotalPrice' }],
+      drilldown: [],
+      groups: [
+        {
+          type: 'fieldValues',
+          field: 'Organization.Name',
+        },
+        {
+          type: 'numberRanges',
+          field: 'LineItem.TotalPrice',
+          ranges: [
+            { from: '0', to: '500' },
+            { from: '500', to: '10000' },
+          ],
+        },
+      ],
+    }
+    let expected = {
+      aggs: {
+        groups: {
+          terms: { field: 'Organization.Name', size: 10 },
+          aggs: {
+            'pivotMetric-sum-LineItem.TotalPrice': {
+              sum: { field: 'LineItem.TotalPrice' },
+            },
+          },
+        },
+      },
+    }
+    let result = await buildQuery(
+      input,
+      testSchemas(['Vendor.City']),
+      () => {} // getStats(search) -> stats(field, statsArray)
+    )
+    expect(result).to.eql(expected)
+  })
+  it('should buildQuery for fieldValues with drilldown and limited depth (deeper)', async () => {
+    let input = {
+      key: 'test',
+      type: 'pivot',
+      values: [{ type: 'sum', field: 'LineItem.TotalPrice' }],
+      drilldown: ['Reno'],
+      groups: [
+        {
+          type: 'fieldValues',
+          field: 'Organization.Name',
+        },
+        {
+          type: 'numberRanges',
+          field: 'LineItem.TotalPrice',
+          ranges: [
+            { from: '0', to: '500' },
+            { from: '500', to: '10000' },
+          ],
+        },
+        {
+          type: 'fieldValues',
+          field: 'Organization.Type',
+        }
+      ],
+    }
+    let expected = {
+      aggs: {
+        pivotFilter: {
+          filter: {
+            bool: { must: [{ term: { 'Organization.Name': 'Reno' } }] },
```

Comment:
I bet this would make it really performant!

Reply:
Yup, that's the idea - the drilldown stuff is actually optimally performant since it's as refined a query as you can get 😎